breadcord/bot.py

```diff
@@ -177,10 +177,20 @@ async def load_wrapper(module_id: str) -> None:
                 return
             await self.modules.get(module_id).load()
 
-        await asyncio.gather(*(
-            load_wrapper(module)
-            for module in self.settings.modules.value
-        ))
+        modules: list[str] = self.settings.modules.value  # type: ignore
+        unduped: list[str] = []
+        for module in modules:
+            if module not in unduped:
+                unduped.append(module)
```

Comment:
Given you're doing the check when iterating anyways, why not warn per duplicate module here as well, and name the module too?
```suggestion
            if module in unduped:
                _logger.warning(f'Duplicate module entry blah blah: {module}')
                continue
            unduped.append(module)
```

Reply:
But why? I was considering it when writing it, but came to the conclusion that warning the user per module serves little to no purpose, It'll be removed either way.

Reply:
Personally, especially with a long module list, I think it's much more helpful to say to the user "hey module X is repeated" rather than "something's repeated but I'm not telling you what". Even if Breadcord tries to fix it for you, I'm not a fan of the lack of transparency. Not a big deal either way though